.ci/test-python-oldest.sh

```diff
@@ -7,9 +7,11 @@
 #
 echo "installing lightgbm's dependencies"
 pip install \
+  'cffi==1.15.1' \
   'dataclasses' \
-  'numpy==1.12.0' \
+  'numpy==1.16.6' \
```

Comment:
Why was this necessary?

Reply:
This is the constraint that `pyarrow==6.0.1` lists on PyPi.

Reply:
I wasn't sure what "on PyPI" meant, so I went looking. Didn't see that in the README at https://pypi.org/project/pyarrow/6.0.1/.

I do see it in the `setup.py` for that version though.

```shell
curl -O https://files.pythonhosted.org/packages/38/c6/97a4133eea642155e7a73cb946d889cadc461a1e6b93f5627af9fdd7b3f3/pyarrow-6.0.1.tar.gz
tar -xvzf ./pyarrow-6.0.1.tar.gz
cat ./pyarrow-6.0.1/setup.py | grep numpy
# 'numpy >= 1.16.6',
```

I think this is acceptable in exchange for testing old versions of `pyarrrow` in this job. `numpy==1.16.6` came out in December 2019 ([link](https://pypi.org/project/numpy/1.16.6/)).

Thanks!

Reply:
> I do see it in the setup.py for that version though.

Yep, that's what I meant :smile: sorry for being imprecise!